src/ophys_etl/modules/segmentation/graph_utils/edge_attributes.py

```diff
@@ -0,0 +1,381 @@
+    for trace in neighbor_traces:
+        masked_neighbors.append(trace[global_mask])
+
+    pearsons = 1.0 - cdist([masked_node0, masked_node1],
+                           masked_neighbors,
+                           metric="correlation")
+
+    return pearsons
+
+
+def add_hnc_gaussian_metric(graph: nx.Graph,
+                            video_path: Path,
+                            neighborhood_radius: Union[int, float],
+                            attribute_name: str = "hnc_Gaussian",
+                            filter_fraction: Optional[float] = None,
+                            full_neighborhood: bool = False) -> nx.Graph():
+    """for each pair of pixels represented by an edge, calculates the Gaussian
+    similarity in correlation space to a neighborhood of size
+    'neighborhood_radius' where the correlation coefficient of all pairs of
+    pixels in that neighborhood are considered.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+        a graph with nodes like (row, col) and edges connecting them
+    video_path: Path
+        path to an hdf5 video file, assumed to have a dataset "data"
+        nframes x nrow x ncol
+    neighborhood_radius: Union[int, float]
+        pixels within <= this radius of the pair centroid are considered
+        in the neighborhood.
+    attribute_name: str
+        name set on each edge for this calculated value
+    filter_fraction: Optional[float]
+        If not None, take the brightest filter_fraction pixels when
+        calculating the Pearson correlation within a neighborhood.
+    full_neighborhood: bool
+        If True, use all of the pixels in the neighborhood when selecting
+        the timesteps to correlate. When False, only use node0 and node1
+        (default = False)
+
+    Returns
+    -------
+    new_graph: nx.Graph
+        an undirected networkx graph, with attribute added to edges
+
+    Notes
+    -----
+    See:
+    "HNCcorr: A Novel Combinatorial Approach for Cell Identification
+    in Calcium-Imaging Movies"
+    Spaen et. al.
+    https://pubmed.ncbi.nlm.nih.gov/31058211/
+
+    """
+
+    with h5py.File(video_path, "r") as f:
+        nrow, ncol = f["data"].shape[1:]
+
+    # find bounds of data to load
+    row_min = np.inf
+    row_max = 0
+    col_min = np.inf
+    col_max = 0
+    for edge in graph.edges:
+        neighborhood = neighborhood_pixels(np.array(edge).mean(axis=0),
+                                           neighborhood_radius,
+                                           nrow,
+                                           ncol)
+        n_mn = neighborhood.min(axis=0)
+        n_mx = neighborhood.max(axis=0)
+        if n_mn[0] < row_min:
+            row_min = n_mn[0]
+        if n_mn[1] < col_min:
+            col_min = n_mn[1]
+        if n_mx[0] > row_max:
+            row_max = n_mx[0]
+        if n_mx[1] > col_max:
+            col_max = n_mx[1]
+
+    # load the data
+    with h5py.File(video_path, "r") as f:
```

Comment:
Same concern, separate out IO. Will making testing easier if you can just pass in an array rather than needing to go create a h5 file.